### PR TITLE
Remove unnecessary poetry.lock file and fix tests in OpenAI SDK cookbook code

### PR DESCRIPTION
**Pull request by smurching**
### Related Issues/PRs
This PR is a follow-up to https://github.com/databricks/genai-cookbook/pull/45

### What changes are proposed in this pull request?
See title; removes an unnecessary large file & fixes tests in OpenAI SDK cookbook sample code

### How is this PR tested?
* Updated unit tests and verified they pass on this PR
* Verified there are no references to the poetry.lock file that's removed in this PR

### CODE REVIEW

File: openai_sdk_agent_app_sample_code/tools/test_code_exec_as_uc_tool.py [hunk 1]
```diff
@@ -1,10 +1,15 @@
 import pytest
 from cookbook.tools.uc_tool import UCTool
+import os
 
 CATALOG = "ep"
 SCHEMA = "cookbook_local_test"
 
 
+if os.getenv("GITHUB_ACTIONS") == "true":
```
Comment (smurching) [inline openai_sdk_agent_app_sample_code/tools/test_code_exec_as_uc_tool.py:9]: As an alternative, we could mock out the WorkspaceClient APIs & DatabricksFunctionClient in these tests, but skipping these in CI for now for simplicity. It also seems important to actually be able to test UC tool execution against a real backend, given that there are currently differences between local & remote execution. If there weren't, I guess we could just test the actual function definition and assume UC execution works fine

File: dev/dev_requirements.txt [hunk 2]
```diff
@@ -2,7 +2,14 @@ jupyter-book
 livereload
 black
 pytest
-mlflow-skinny[databricks]
```
Comment (epec254) [inline dev/dev_requirements.txt:5]: I don't think this is needed?  at least in the openai SDK folder, I was using poetry to install these locally and had mirrored that to the requirements.txt files there...

File: openai_sdk_agent_app_sample_code/tools/test_code_exec_as_uc_tool.py (hunk 1 above)
Comment (epec254) [reply openai_sdk_agent_app_sample_code/tools/test_code_exec_as_uc_tool.py:9]: +1

**Review by epec254 — APPROVED** [review]

File: dev/dev_requirements.txt (hunk 2 above)
Comment (smurching) [reply dev/dev_requirements.txt:5]: Good point, this is the combined requirements that are installed in CI - but it makes sense to actually mirror user-facing env setup steps in CI (i.e. cd into `openai_sdk_agent_app_sample_code` and install dependencies via `poetry`). I'm hoping we can still do that effectively without checking in `poetry.lock`, unless that buys us better reproducibility than installing python libs via `requirements.txt`

**Pull request merged**